### PR TITLE
improve observability

### PR DESCRIPTION
**Pull request by tx19980520**
1. add prom basic architecture
2. add jaeger basic feature

### CODE REVIEW

**Comment by tx19980520** [comment]
you can see the performance about [simple](http://106.15.225.249:16686/trace/6fb82fca5d269124) and [if-else](http://106.15.225.249:16686/trace/5545f1a28cbea93c)

**Comment by tx19980520** [comment]
ubuntu@tass:~/scheduler$ sudo go test -p 1 -count=1 ./...
?       github.com/tass-io/scheduler    [no test files]
?       github.com/tass-io/scheduler/cmd        [no test files]
?       github.com/tass-io/scheduler/examples/middleware        [no test files]
?       github.com/tass-io/scheduler/pkg/dto    [no test files]
?       github.com/tass-io/scheduler/pkg/env    [no test files]
?       github.com/tass-io/scheduler/pkg/event  [no test files]
?       github.com/tass-io/scheduler/pkg/event/initial  [no test files]
?       github.com/tass-io/scheduler/pkg/event/metrics  [no test files]
?       github.com/tass-io/scheduler/pkg/event/qps      [no test files]
ok      github.com/tass-io/scheduler/pkg/event/schedule 2.546s
?       github.com/tass-io/scheduler/pkg/event/source   [no test files]
?       github.com/tass-io/scheduler/pkg/event/ttl      [no test files]
?       github.com/tass-io/scheduler/pkg/http   [no test files]
?       github.com/tass-io/scheduler/pkg/http/controller        [no test files]
ok      github.com/tass-io/scheduler/pkg/initial        5.018s
?       github.com/tass-io/scheduler/pkg/initial/wrapper        [no test files]
?       github.com/tass-io/scheduler/pkg/initial/wrapper/plugin [no test files]
?       github.com/tass-io/scheduler/pkg/middleware     [no test files]
?       github.com/tass-io/scheduler/pkg/middleware/initial     [no test files]
?       github.com/tass-io/scheduler/pkg/middleware/lsds        [no test files]
?       github.com/tass-io/scheduler/pkg/middleware/qps [no test files]
?       github.com/tass-io/scheduler/pkg/middleware/static      [no test files]
?       github.com/tass-io/scheduler/pkg/prom   [no test files]
?       github.com/tass-io/scheduler/pkg/runner [no test files]
ok      github.com/tass-io/scheduler/pkg/runner/fnscheduler     2.580s
?       github.com/tass-io/scheduler/pkg/runner/helper  [no test files]
ok      github.com/tass-io/scheduler/pkg/runner/instance        20.133s
ok      github.com/tass-io/scheduler/pkg/runner/lsds    1.580s
?       github.com/tass-io/scheduler/pkg/runner/ttl     [no test files]
?       github.com/tass-io/scheduler/pkg/schedule       [no test files]
?       github.com/tass-io/scheduler/pkg/span   [no test files]
?       github.com/tass-io/scheduler/pkg/store  [no test files]
?       github.com/tass-io/scheduler/pkg/tools/base64   [no test files]
?       github.com/tass-io/scheduler/pkg/tools/common   [no test files]
?       github.com/tass-io/scheduler/pkg/tools/errorutils       [no test files]
ok      github.com/tass-io/scheduler/pkg/tools/k8sutils 0.064s
?       github.com/tass-io/scheduler/pkg/tools/log      [no test files]
?       github.com/tass-io/scheduler/pkg/tools/register [no test files]
?       github.com/tass-io/scheduler/pkg/trace  [no test files]
ok      github.com/tass-io/scheduler/pkg/workflow       2.084s
?       github.com/tass-io/scheduler/test       [no test files]
ok      github.com/tass-io/scheduler/test/cross 21.115s
ok      github.com/tass-io/scheduler/test/initial       14.644s
ok      github.com/tass-io/scheduler/test/metrics       25.222s
ok      github.com/tass-io/scheduler/test/pipe  4.529s
ok      github.com/tass-io/scheduler/test/switch        2.545s

File: pkg/span/span.go [hunk 1]
```diff
@@ -1,10 +1,149 @@
...
+			return
+		}
+		if spanName == "" {
+			spanName = span.flowName
+		}
+		span.sp = opentracing.StartSpan(spanName, opentracing.ChildOf(span.parent))
+	})
+}
+
+// luanlunlaide
```
Comment (linxuyalun) [inline pkg/span/span.go:118]: 。。。

File: pkg/span/span.go [hunk 2]
```diff
@@ -1,10 +1,149 @@
...
+	if span.sp == nil {
+		// todo check
+		return
+	}
+	span.finishOnce.Do(func() {
+		span.sp.Finish()
+	})
+}
+
+func (span *Span) InjectRoot(header http.Header) {
```
Comment (linxuyalun) [inline pkg/span/span.go:140]: 没太看懂这个函数的含义

**Comment by tx19980520** [comment]
ubuntu@tass:~/scheduler$ sudo go test -p 1 -count=1 ./...
?       github.com/tass-io/scheduler    [no test files]
?       github.com/tass-io/scheduler/cmd        [no test files]
?       github.com/tass-io/scheduler/examples/middleware        [no test files]
?       github.com/tass-io/scheduler/pkg/dto    [no test files]
?       github.com/tass-io/scheduler/pkg/env    [no test files]
?       github.com/tass-io/scheduler/pkg/event  [no test files]
?       github.com/tass-io/scheduler/pkg/event/initial  [no test files]
?       github.com/tass-io/scheduler/pkg/event/metrics  [no test files]
?       github.com/tass-io/scheduler/pkg/event/qps      [no test files]
ok      github.com/tass-io/scheduler/pkg/event/schedule 2.540s
?       github.com/tass-io/scheduler/pkg/event/source   [no test files]
?       github.com/tass-io/scheduler/pkg/http   [no test files]
?       github.com/tass-io/scheduler/pkg/http/controller        [no test files]
ok      github.com/tass-io/scheduler/pkg/initial        4.393s
?       github.com/tass-io/scheduler/pkg/initial/wrapper        [no test files]
?       github.com/tass-io/scheduler/pkg/initial/wrapper/plugin [no test files]
?       github.com/tass-io/scheduler/pkg/middleware     [no test files]
?       github.com/tass-io/scheduler/pkg/middleware/initial     [no test files]
?       github.com/tass-io/scheduler/pkg/middleware/lsds        [no test files]
?       github.com/tass-io/scheduler/pkg/middleware/qps [no test files]
?       github.com/tass-io/scheduler/pkg/middleware/static      [no test files]
?       github.com/tass-io/scheduler/pkg/prom   [no test files]
?       github.com/tass-io/scheduler/pkg/runner [no test files]
ok      github.com/tass-io/scheduler/pkg/runner/fnscheduler     2.558s
?       github.com/tass-io/scheduler/pkg/runner/helper  [no test files]
ok      github.com/tass-io/scheduler/pkg/runner/instance        20.825s
ok      github.com/tass-io/scheduler/pkg/runner/lsds    1.571s
?       github.com/tass-io/scheduler/pkg/runner/ttl     [no test files]
?       github.com/tass-io/scheduler/pkg/schedule       [no test files]
?       github.com/tass-io/scheduler/pkg/span   [no test files]
?       github.com/tass-io/scheduler/pkg/store  [no test files]
?       github.com/tass-io/scheduler/pkg/tools/base64   [no test files]
?       github.com/tass-io/scheduler/pkg/tools/common   [no test files]
?       github.com/tass-io/scheduler/pkg/tools/errorutils       [no test files]
ok      github.com/tass-io/scheduler/pkg/tools/k8sutils 0.076s
?       github.com/tass-io/scheduler/pkg/tools/log      [no test files]
?       github.com/tass-io/scheduler/pkg/tools/register [no test files]
?       github.com/tass-io/scheduler/pkg/trace  [no test files]
ok      github.com/tass-io/scheduler/pkg/workflow       2.064s
?       github.com/tass-io/scheduler/test       [no test files]
ok      github.com/tass-io/scheduler/test/cross 21.244s
ok      github.com/tass-io/scheduler/test/initial       15.495s
ok      github.com/tass-io/scheduler/test/metrics       25.214s
ok      github.com/tass-io/scheduler/test/pipe  4.536s
ok      github.com/tass-io/scheduler/test/switch        2.499s

**Review by linxuyalun — APPROVED** [review]

**Pull request merged**